modules/ui.py

```diff
@@ -429,18 +430,132 @@ def refresh(request: gradio.routes.Request):
     return refresh_button
 
 
-def create_upload_button(label, elem_id, destination_dir):
-    def upload(file):
+def create_upload_button(label, elem_id, destination_dir, model_tracking_csv="models.csv"):

```

Comment:
Both destination_dir and model_tracking_csv sound like global variables to me. But I am okay leaving them here.

Reply:
We need them because there are multiple locations that need upload buttons. I need to keep it so different models will be put into different folders.